### ISSUE TITLE
Inconsistent comparison between empty string and None/NA

### ISSUE THREAD

**Issue by 153957** [maxbachmann/rapidfuzz#387]
There appears to be a bug in how empty strings, `None`, and `pandas.NA` are handled when they are either the first or second argument:

```python
>>> cdist(['example1', '', None, pandas.NA], ['example1', '', None, pandas.NA])
array([[100.,   0.,   0.,   0.],
       [  0., 100.,   0.,   0.],
       [  0., 100.,   0.,   0.],
       [  0., 100.,   0.,   0.]], dtype=float32)
```
(also tested with `DamerauLevenshtein.normalized_similarity` as scorer, same inconsistency)
I would expect the result to be symmetric.

Note, when the array is stored in a variable rapidfuzz uses the 'single_list' function which makes the result symmetric, with the result scores that I would expect.

```python
>>> data = ['example1', '', None, pandas.NA]
>>> cdist(data, data)
array([[100.,   0.,   0.,   0.],
       [  0., 100.,   0.,   0.],
       [  0.,   0.,   0.,   0.],
       [  0.,   0.,   0.,   0.]], dtype=float32)
```

Directly calling a scorer gives the expected results:
```python
>>> DamerauLevenshtein.normalized_similarity(None, '')
0.0
>>> DamerauLevenshtein.normalized_similarity('', None)
0.0
```

**Comment by maxbachmann** [comment]
It appears to work fine on my machine:
```python
>>> from rapidfuzz.process import cdist
>>> import pandas
>>> cdist(['example1', '', None, pandas.NA], ['example1', '', None, pandas.NA])
array([[100.,   0.,   0.,   0.],
       [  0., 100.,   0.,   0.],
       [  0.,   0.,   0.,   0.],
       [  0.,   0.,   0.,   0.]], dtype=float32)
```

Are you using the latest version of rapidfuzz:
```python
>>> rapidfuzz.__version__
'3.9.2'
```
And on which platform + Python version?

**Comment by 153957** [comment]
On my Apple Silicon (ARM) Mac in a Python 3.12 Docker container.
```
$ python --version
Python 3.12.2
$ pip freeze | grep rapid
rapidfuzz==3.9.1
```
Just upgraded to Rapidfuzz 3.9.2, and I can still reproduce the same issue.

Let me know if you'd like me to run any other tests.

**Comment by maxbachmann** [comment]
I had a look into the different code paths. There was a bug in the non SIMD branch. This should be fixed in `v3.9.3`

**Comment by 153957** [comment]
Thank you! it works as expected now.
